changelog/1025.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Fixed a bug that prevented nested iterations of a single
+`~plasmapy.particles.IonizationState` or
+`~plasmapy.particles.IonizationStateCollection` instance.
```

Comment:
Also, due to your commit message I'm now trying to find a way to neatly put a joke about how this is not the greatest and best PR in the world; this is [just a changelog](https://www.youtube.com/watch?v=xy5S8s6rTH0). But I ain't even had tea yet, so, that can wait.

BTW, this PR was one of my top 10 things to wake up to! ;)